src/adapter.c

```diff
@@ -834,6 +835,13 @@ void close_adapter_for_stream(int sid, int aid)
 		mark_pids_deleted(aid, sid, NULL);
 	update_pids(aid);
 	adapter_update_threshold(ad);
+	if (stby && !is_slave && ad->type == ADAPTER_SATIP && ad->sid_cnt == 0)
+	{
+		LOG("adapter %d closing for standby", ad->id);
+		ad->is_standby = 1;
+		close_adapter(ad->id);
```

Comment:
Calling close adapter will close the entire adapter and close the socket.

Reply:
> Calling close adapter will close the entire adapter and close the socket.

Here I don't accept it! Please, review your previous comment. If the `ad->is_standby` flag is set, then the call to `close_adapter()` doesn't close the adapter. It only sends the TEARDOWN. Please, note that this code is included in a test with `ad->type == ADAPTER_SATIP`. So only if the adapter can check the `is_standby` flag, then the function `close_adapter()` is called.

I'm sure this part of the code is correct. I tested and debugged it. No problem at all. The TEARDOWN command is sended and the adapter continues open, even the socket.